pennylane/ops/qubit/state_preparation.py

```diff
@@ -89,3 +89,34 @@ def decomposition(state, wires):
 
     def adjoint(self):
         raise qml.ops.AdjointError("No adjoint exists for QubitStateVector operations.")
+
+
+class QubitDensityMatrix(Operation):
+    r"""QubitDensityMatrix(state, wires)
+    Prepare subsystems using the given density matrix.
+    If not all the wires are specified, remaining dimension is filled by :math:`\mathrm{tr}_{in}(\rho)`,
+    where :math:`\rho` is the full system density matrix before this operation and :math:`\mathrm{tr}_{in}` is a
+    partial trace over the subsystem to be replaced by input state.
+
+    **Details:**
+
+    * Number of wires: Any (the operation can act on any number of wires)
+    * Number of parameters: 1
+    * Gradient recipe: None
+
+    .. note::
+
+        Exception raised if the ``QubitDensityMatrix`` operation is not supported natively on the
+        target device.
+
+    Args:
+        state (array[complex]): a density matrix of size (2**len(wires), 2**len(wires))
+        wires (Sequence[int] or int): the wire(s) the operation acts on
+    """
```

Comment:
It would be worth adding an example here so users who look at the docs know how to use this functionality ! 

```suggestion
    r"""QubitDensityMatrix(state, wires)
    Prepare subsystems using the given density matrix.
    If not all the wires are specified, remaining dimension is filled by :math:`\mathrm{tr}_{in}(\rho)`,
    where :math:`\rho` is the full system density matrix before this operation and :math:`\mathrm{tr}_{in}` is a
    partial trace over the subsystem to be replaced by input state.

    **Details:**

    * Number of wires: Any (the operation can act on any number of wires)
    * Number of parameters: 1
    * Gradient recipe: None

    .. note::

        Exception raised if the ``QubitDensityMatrix`` operation is not supported natively on the
        target device.

    Args:
        state (array[complex]): a density matrix of size (2**len(wires), 2**len(wires))
        wires (Sequence[int] or int): the wire(s) the operation acts on

    .. UsageDetails::
        ** Example: **
        
        .. code-block:: python
            import pennylane as qml
            
            nr_wires = 2
            rho = np.zeros((2 ** nr_wires, 2 ** nr_wires), dtype=np.complex128)
            rho[0, 0] = 1  # initialize the pure state density matrix for the |0><0| state
            dev = qml.device("default.mixed", wires=2) 
            
            @qml.qnode(dev)
            def circuit():
                qml.QubitDensityMatrix(initialize_state, wires=[0, 1])
                 return qml.state()
            
        Running this circuit: 
        
        >>>circuit()
        [[1.+0.j 0.+0.j 0.+0.j 0.+0.j]
         [0.+0.j 0.+0.j 0.+0.j 0.+0.j]
         [0.+0.j 0.+0.j 0.+0.j 0.+0.j]
         [0.+0.j 0.+0.j 0.+0.j 0.+0.j]]
    """
```